truffleruby/src/main/c/cext/ruby.c

```diff
@@ -379,15 +379,15 @@ int NUM2INT(VALUE value) {
 }
 
 unsigned int NUM2UINT(VALUE value) {
-  return (unsigned int) truffle_invoke_i(RUBY_CEXT, "NUM2UINT", value);
+  return (unsigned int) truffle_invoke_l(RUBY_CEXT, "NUM2LONG", value);
 }
 
 long NUM2LONG(VALUE value) {
   return truffle_invoke_l(RUBY_CEXT, "NUM2LONG", value);
 }
 
 unsigned long rb_num2ulong(VALUE val) {
-  rb_tr_error("rb_num2ulong not implemented");
+  return (unsigned long)truffle_invoke_l(RUBY_CEXT, "rb_num2ulong", val);
```

Comment:
We pass back long from Java containing the appropriate bits, and the explicit cat doesn't change it. All a bit sneaky as Java doesn't have unsigned types for us to use.